Setting a custom registry throws an error when installing

**Describe the bug**

Hello! 👋 

I'd like to use this action to publish a package to the GitHub packages registry. When using the config snippet below, I get this error when running semantic release:

```
Error: Command failed: npm ci --only=prod
npm ERR! code E401
npm ERR! 401 Unauthorized - GET https://npm.pkg.github.com/@actions%2fcore - Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2020-09-18T13_12_06_037Z-debug.log

    at ChildProcess.exithandler (child_process.js:295:12)
    at ChildProcess.emit (events.js:210:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5) {
  killed: false,
  code: 1,
  signal: null,
  cmd: 'npm ci --only=prod',
  stdout: '',
  stderr: 'npm ERR! code E401\n' +
    'npm ERR! 401 Unauthorized - GET https://npm.pkg.github.com/@actions%2fcore - Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.\n' +
    '\n' +
    'npm ERR! A complete log of this run can be found in:\n' +
    'npm ERR!     /home/runner/.npm/_logs/2020-09-18T13_12_06_037Z-debug.log\n'
}
```

In addition, the process continues on with the rest of the workflow when this error is encountered. I think it might be nice to call `process.exit(1)` if we can't recover.

**Workflow**

Rest of this snipped:

```yml
- name: Semantic Release
  uses: cycjimmy/semantic-release-action@v2
  env:
    GITHUB_TOKEN: ${{ secrets.PERSONAL_TOKEN }}
    NPM_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    NPM_CONFIG_REGISTRY: https://npm.pkg.github.com
  with:
    semantic_version: 17
    dry_run: false
    branch: main
    extra_plugins: |
      @semantic-release/commit-analyzer
      @semantic-release/release-notes-generator
      @semantic-release/changelog
      @semantic-release/git
      @semantic-release/github
      @semantic-release/npm
```

**Expected behavior**

I expected the action to not throw when using the custom registry.

**Additional context**

I believe this is a limitation with the GitHub packages registry and scoped packages, you can see more details here:

https://gist.github.com/azu/31530916cbce0fd2fc1f4d8f6cf0fae1

I found a solution which was to replace the `NPM_CONFIG_REGISTRY` variable with a `publishConfig` in my `package.json`, like so:

```json
{
    "publishConfig": {
        "registry": "https://npm.pkg.github.com"
    }
}
```

Would you accept a pull request adding a section on private packages to the readme?

Thanks!

--- a/README.md
+++ b/README.md
@@ -28,6 +28,19 @@ steps:
 **IMPORTANT**: `GITHUB_TOKEN` does not have the required permissions to operate on protected branches.
 If you are using this action for protected branches, replace `GITHUB_TOKEN` with [Personal Access Token](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line). If using the `@semantic-release/git` plugin for protected branches, avoid persisting credentials as part of `actions/checkout@v2` by setting the parameter `persist-credentials: false`. This credential does not have the required permission to operate on protected branches.
 
+#### Private Packages
+
+If you are using this action to publish to the npm [GitHub Packages Registry][github-packages-registry],
+then make sure that you configure this in your `package.json` file:
+
+```json
+{
+    "publishConfig": {
+        "registry": "https://npm.pkg.github.com"
+    }
+}
+```
+
 ### Inputs
 | Input Parameter  | Required | Description |
 |:----------------:|:--------:|-------------|
@@ -244,3 +257,4 @@ This project is released under the [MIT License][license-url].
 
 [changelog-url]: https://github.com/cycjimmy/semantic-release-action/blob/master/docs/CHANGELOG.md
 
+[github-packages-registry]: https://github.com/features/packages

